akka-actor/src/main/scala/akka/pattern/BackoffOnRestartSupervisor.scala

```diff
@@ -67,9 +80,11 @@ private class BackoffOnRestartSupervisor(
     case Terminated(`childRef`) ⇒
       become(receive)
       child = None
-      val restartDelay = BackoffSupervisor.calculateDelay(restartCount, minBackoff, maxBackoff, randomFactor)
-      context.system.scheduler.scheduleOnce(restartDelay, self, BackoffSupervisor.StartChild)
-      restartCount += 1
+      if (maxRestartAttempts == -1 || restartCount < maxRestartAttempts) {
```

Comment:
how is this related to `if (strategy.maxNrOfRetries >= 0 && nextRestartCount > strategy.maxNrOfRetries)` a few lines up?

Do we have two max values in play now?

Reply:
I've realized it's existence before, but changing the maxNrOfRetries of the strategy which has been set was a little confusing to me. I should've asked first. What do you propose? should I rewrite it using the maxNrOfRetries of the strategy?

Reply:
I think it can be confusing to have two different competing max. Could we take a step back and list all API methods and see where we should add it.

* BackoffSupervisor.props - here we could add an overloaded method that takes the additional maxNrOfRetries
* BackoffSupervisor.propsWithSupervisorStrategy - if the strategy is given then we should not have another parameter for maxNrOfRetries, instead that is taken from the strategy
* Backoff.onFailure - add an overloaded method that takes the additional maxNrOfRetries
* Backoff.onStop - add an overloaded method that takes the additional maxNrOfRetries
* BackoffOptions.withSupervisorStrategy - if this is used then it will replace previously defined maxNrOfRetries if any (must be documented)
* BackoffOptions.withDefaultStoppingStrategy - don't replace previously defined maxNrOfRetries if any

Did I miss any other way to create the backoff supervisor?

Reply:
I think that's all. 
Thank you.

Reply:
I have implemented it the way we discussed. Could you review it again?